permission_handler_web/lib/web_delegate.dart

```diff
@@ -0,0 +1,224 @@
+    return _toPermissionStatus(webPermissionStatus?.state);
+  }
+
+  Future<bool> _requestMicrophonePermission(html.MediaDevices devices) async {
+    html.MediaStream? mediaStream;
+
+    try {
+      mediaStream = await devices.getUserMedia({'audio': true});
+
+      // In browsers, calling [getUserMedia] will start the recording
+      // automatically right after. This is undesired behavior as
+      // [requestPermission] is expected to request permission only.
+      //
+      // The manual stop action is then needed here for to stop the automatic
+      // recording.
+
+      if (mediaStream.active!) {
+        final audioTracks = mediaStream.getAudioTracks();
+        if (audioTracks.isNotEmpty) {
+          audioTracks[0].stop();
+        }
+      }
+    } on html.DomException {
+      return false;
+    }
+
+    return true;
+  }
+
+  Future<bool> _requestCameraPermission(html.MediaDevices devices) async {
+    html.MediaStream? mediaStream;
+
+    try {
+      mediaStream = await devices.getUserMedia({'video': true});
+
+      // In browsers, calling [getUserMedia] will start the recording
+      // automatically right after. This is undesired behavior as
+      // [requestPermission] is expected to request permission only.
+      //
+      // The manual stop action is then needed here for to stop the automatic
+      // recording.
+
+      if (mediaStream.active!) {
+        final videoTracks = mediaStream.getVideoTracks();
+        if (videoTracks.isNotEmpty) {
+          videoTracks[0].stop();
+        }
+      }
+    } on html.DomException {
+      return false;
+    }
+
+    return true;
+  }
+
+  Future<bool> _requestNotificationPermission() async {
+    bool granted = false;
+    html.Notification.requestPermission().then((permission) => {
+          if (permission == "granted") {granted = true}
+        });
+
+    return granted;
+  }
+
+  Future<bool> _requestLocationPermission(html.Geolocation geolocation) async {
+    try {
+      await geolocation.getCurrentPosition();
+      return true;
+    } on html.PositionError {
+      return false;
+    }
+  }
+
+  Future<PermissionStatus> _requestSingularPermission(
+      Permission permission) async {
+    bool permissionGranted = false;
+
+    switch (permission) {
+      case Permission.microphone:
+        permissionGranted = await _requestMicrophonePermission(devices!);
+        break;
+      case Permission.camera:
+        permissionGranted = await _requestCameraPermission(devices!);
+        break;
+      case Permission.notification:
+        permissionGranted = await _requestNotificationPermission();
+        break;
+      case Permission.location:
+        permissionGranted = await _requestLocationPermission(geolocation!);
+        break;
+      default:
+        throw UnimplementedError(
+          '_requestSingularPermission() has not been implemented for '
+          '${permission.toString()} on web.',
+        );
```

Comment:
I propose to let this error bubble up and let developers handle it accordingly. This way they will be actively informed that they are requesting permissions that are not supported. 

Since developers will see this error it would be good to throw a `UnsupportedError` instead and update the message so it doesn't refer to any internal implementation details. For example:
```suggestion
        throw UnsupportedError(
          'The ${permission.toString()} permission is currently not supported on web.',
        );
```